perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java

```diff
@@ -1164,11 +1171,46 @@ public User getUserFromMessage(PerunSessionImpl sess, String message) throws Int
 		return new Pair<>(number, unit);
 	}
 
+	/**
+	 * Extracts expiration of the given certificates.
+	 *
+	 * @param certificates as a map where the key is a DN and the value is a certificate
+	 * @return map where the key is certificate DN and the value is a certificate expiration
+	 */
+	public static Map<String, String> retrieveCertificatesExpiration(Map<String, String> certificates) {
+		Utils.notNull(certificates, "certificates");
+		Map<String, String> resultMap = new LinkedHashMap<>();
+		certificates.forEach((key, value) -> resultMap.put(key, getCertificateExpiration(value)));
+		return resultMap;
+	}
+
 	public PerunBl getPerunBl() {
 		return this.perunBl;
 	}
 
 	public void setPerunBl(PerunBl perunBl) {
 		this.perunBl = perunBl;
 	}
+
+	/**
+	 * Retrieve expiration of the given certificate
+	 *
+	 * @param certificate in PEM format from which the expiration will be retrieved
+	 * @return Certificate expiration as String
+	 */
+	private static String getCertificateExpiration(String certificate) {
+		String certWithoutBegin = certificate.replaceFirst("-----BEGIN CERTIFICATE-----", "");
+		String rawCert = certWithoutBegin.replaceFirst("-----END CERTIFICATE-----", "");
```

Comment:
You should test not-null on certificate variable.

Reply:
Added